client/gutenberg/editor/controller.js

```diff
@@ -194,3 +194,9 @@ export const post = ( context, next ) => {
 
 	return next();
 };
+
+export const siteEditor = ( context, next ) => {
+	context.primary = <CalypsoifyIframe postType={ 'site' } />;
```

Comment:
I am not sure if we should add any other props to this as seen in the original `CalypsoifyIframe` above it.
Im not sure what `pressThis` is but maybe something we would want?  Everything else seems like it is either specific to dotcom FSE, post editing, or jetpack support for post editing.

Reply:
Yeah, this seems like enough for me! I think the only thing I might add is the `key`, but it really isn't needed. We only needed it in the other one so that the iFrame would entirely re-mount in React if the post ID changed. But maybe we could add it "just in case" here set to the site ID?

Reply:
Oh thats interesting.  Why would the postId change?  And I would have expected that to force a re-render through the changing of the `postId` prop as well anyways.

Reply:
It does do a re-render, but the subtle difference is that React doesn't reinitialize the component. So all of the things that happen as the component is mounting for the first time wouldn't happen again. I wasn't quite able to track down exactly why it didn't work on re-render, but I think it might have to do with the lifecycle events, like `componentWillMount`. At any rate, it didn't cause the  `iframe` element to update with the new URL, so using `key` [forces it to create a new component entirely](https://reactjs.org/blog/2018/06/07/you-probably-dont-need-derived-state.html#recommendation-fully-uncontrolled-component-with-a-key).

Reply:
Oh interesting!  I wonder if the `componentWillUnmount` was the culprit in that case, either way I think adding the `siteId` here will make sense as a precaution as long as we make a comment of it.